aiobotocore/httpsession.py

```diff
@@ -93,25 +92,42 @@ def __init__(
         # it also pools by host so we don't need a manager, and can pass proxy via
         # request so don't need proxy manager
 
-        ssl_context = None
+        context = None
         if bool(verify):
+            context = self._get_ssl_context()
+            if self._cert_file:
+                context.load_cert_chain(self._cert_file, self._key_file)
+
+            # inline self._setup_ssl_cert
```

Comment:
I'd prefer we kept this logic in `_setup_proxy_ssl_context` as it'll be easier to diff going forwards.  There's a flaw in this approach though because the proxy context depends on if the host we're connecting to is an IPV4 address or not: https://github.com/boto/botocore/blob/1.34.23/botocore/httpsession.py#L396-L397.  I think we need to implement a context per proxy url unfortunately.

Reply:
which means a tcp connector per proxy url I think :(

Reply:
> which means a tcp connector per proxy url I think :(

I'm afraid, you're right. I'll try to address that in an updated PR.

Reply:
sorry :]